bundles/org.eclipse.mita.program.ui/src/org/eclipse/mita/program/ui/contentassist/ProgramDslProposalProvider.xtend

```diff
@@ -132,10 +144,110 @@ class ProgramDslProposalProvider extends AbstractProgramDslProposalProvider {
 	override getPriorityHelper() {
 		priorityHelper
 	}
+	
+	def proposeComplexTypeConstructors(Boolean scoped, EObject model, ContentAssistContext context, ICompletionProposalAcceptor acceptor) {
+		val scope = scopeProvider.getScope(model, ExpressionsPackage.eINSTANCE.elementReferenceExpression_Reference);
+		for (element : scope.allElements) {
+			val obj = element.EObjectOrProxy;
+			val fromPlatform = EcoreUtil2.getContainerOfType(obj, Program) === null;
```

Comment:
I would prefer a positive check, i.e. `EcoreUtil2.getContainerOfType(obj, *Platform*) !== null`

Reply:
I would like that more as well, however if the platform isn't fully resolved, `SystemSpecification` might have no eContainer.

Reply:
Ok I see.